### ISSUE TITLE
Invariant Violation: Bad output range 

### ISSUE THREAD

**Issue by AliaMYH** [react-native-community/react-native-tab-view#478]
### Current behaviour

When I try and render my tab-view, I get the following error:

```
Invariant Violation: Bad output range

This error is located at:
    in TabBar (at Pager.js:106)
    in RCTView (at View.js:71)
    in View (at TabViewAnimated.js:194)
    in TabViewAnimated (at Pager.js:148)
    in Pager (created by Connect(Pager))
    in Connect(Pager) (at Subcategories.js:48)
    in RCTView (at View.js:71)
    in View (at Subcategories.js:102)
    in RCTView (at View.js:71)
    in View (at Subcategories.js:75)
    in Subcategories (created by Connect(Subcategories))
    in Connect(Subcategories) (at SceneView.js:17)
    in SceneView (at DrawerScreen.js:21)
    in DrawerScreen (at withCachedChildNavigation.js:44)
    in withCachedChildNavigation(DrawerScreen) (at DrawerNavigator.js:71)
    in Unknown (at createNavigator.js:13)
    in Navigator (at DrawerView.js:154)
    in RCTView (at View.js:71)
    in View (at DrawerLayoutAndroid.android.js:186)
    in AndroidDrawerLayout (at DrawerLayoutAndroid.android.js:200)
    in DrawerLayoutAndroid (at DrawerView.js:134)
    in DrawerView (at DrawerNavigator.js:88)
    in Unknown (at createNavigator.js:13)
    in Navigator (at createNavigationContainer.js:214)
    in NavigationContainer (at SceneView.js:17)
    in SceneView (at CardStack.js:385)
    in RCTView (at View.js:71)
    in View (at createAnimatedComponent.js:147)
    in AnimatedComponent (at Card.js:12)
    in Card (at PointerEventsContainer.js:39)
    in Container (at CardStack.js:414)
    in RCTView (at View.js:71)
    in View (at CardStack.js:348)
    in RCTView (at View.js:71)
    in View (at CardStack.js:347)
    in CardStack (at CardStackTransitioner.js:66)
    in RCTView (at View.js:71)
    in View (at Transitioner.js:142)
    in Transitioner (at CardStackTransitioner.js:19)
    in CardStackTransitioner (at StackNavigator.js:38)
    in Unknown (at createNavigator.js:13)
    in Navigator (at createNavigationContainer.js:214)
    in NavigationContainer (at App.js:99)
    in Provider (at App.js:113)
    in App (at renderApplication.js:35)
    in RCTView (at View.js:71)
    in View (at AppContainer.js:102)
    in RCTView (at View.js:71)
    in View (at AppContainer.js:122)
    in AppContainer (at renderApplication.js:34)
```
My route only has one page:

`{key: "3", title: "Milk"}
`
Is the problem that the key is not zero-based or one-based?

### Expected behaviour

This is the first render of the tab-view, so there isn't a chance for any index changes yet.

### Code sample

```
const initialLayout = {
  height: 0,
  width: Dimensions.get('window').width,
};

type State = NavigationState<
  Route<{
    key: string,
    title: string,
  }>
>;

class Pager extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      index: 0,
      routes: [
        // { key: '1', title: 'Fruit' },
        // { key: '2', title: 'Vegetables' },
        // { key: '3', title: 'Fresh Herbs' },
        // { key: '4', title: 'Other' },
      ],
    };
    console.log('before filling routes');
    console.log( this.props.currentSubcategories );

    for ( var subcat in this.props.currentSubcategories)
    {
      console.log('while filling routes');
      this.state.routes.push({ 'key': this.props.currentSubcategories[subcat].id.toString(), 'title': this.props.currentSubcategories[subcat].title });
    }
    console.log(this.state.routes);


  }



  _handleIndexChange = index => {
    this.props.subcategoryChosen(this.state.routes[this.state.index].key);
    this.props.getProducts(this.props.chosenSubcategory);
    this.setState({ index });

  }

  _renderLabel = props => ({ route, index }) => {
    const inputRange = props.navigationState.routes.map((x, i) => i);
    const outputRange = inputRange.map(
      inputIndex => (inputIndex === index ? '#24572c' : 'grey')
    );
    const color = props.position.interpolate({
      inputRange,
      outputRange,
    });

    return (
      <Animated.Text style={[styles.label, { color }]}>
        {route.title}
      </Animated.Text>
    );
  };

  _renderHeader = props => (
    <TabBar
      {...props}
      pressColor="rgba(255, 64, 129, .5)"
      onTabPress={this._handleTabItemPress}
      renderLabel={this._renderLabel(props)}
      indicatorStyle={styles.indicator}
      tabStyle={styles.tab}
      style={styles.headers}
      scrollEnabled= {true}
    />
  );

  _renderScene = ({ route }) => {

    return (
      <ItemList assignment = {route.key} style={[ styles.container ]} />
    );
   
  };

  render() {
    console.log('Rendering Pager');
    return (
      <TabViewAnimated
        style={styles.container}
        navigationState={this.state}
        renderScene={this._renderScene}
        renderHeader={this._renderHeader}
        onIndexChange={this._handleIndexChange}
        initialLayout={initialLayout}
      />
    );
  }
}
```

### Your Environment

| software              | version
| --------------------- | -------
| ios or android        | android 7
| react-native          | 0.53.3
| react-native-tab-view | 0.0.74
| node                  |
| npm or yarn           | nom 5.6.0

**Comment by ghoshabhi** [comment]
@AliaMYH : Do you get this error when using storing routes statically inside the `state` ?

**Comment by AliaMYH** [comment]
@ghoshabhi  I'll try it out and tell you. But the situation is that I have multiple pages of tab-view, depending on variables retrieved from API calls. 

I have two pages: One loads fine, where the key values are '1' and '2' and the contents of the scene load without a problem, the other page, which is the one i'm talking about above, gives me that error when I try to click on it. Which is why i thought maybe the key is the issue.

**Comment by ghoshabhi** [comment]
Oh I see you're also animating the label component which depends on the `props` received - which in your case are coming from a `setState` function call. Since setState is async, it might be possible that you don't have the routes available in the state before your tabview component started rendering.

Try to have a check to see if `this.state.routes` is empty or not on the first render.

**Comment by satya164** [comment]
Can provide a runnable example on snack please?

**Comment by satya164** [comment]
Hey, I just released a new alpha `2.0.0-alpha.0` of the library. It's rewritten using [react-native-gesture-handler](https://github.com/kmagiera/react-native-gesture-handler) and [react-native-reanimated](https://github.com/kmagiera/react-native-reanimated) addresses a many platform specific bugs and performance problems. The documentation is updated as well.

Please try the new version and see if it addresses your issue. If not, please open a new issue following the issue template.
